src/EventTickets/Routes/GetEventsListTable.php

```diff
@@ -0,0 +1,143 @@
+                        'default' => 1,
+                        'minimum' => 1
+                    ],
+                    'perPage' => [
+                        'type' => 'integer',
+                        'default' => 30,
+                        'minimum' => 1
+                    ],
+                    'search' => [
+                        'type' => 'string',
+                        'required' => false,
+                        'sanitize_callback' => 'sanitize_text_field',
+                    ],
+                    'sortColumn' => [
+                        'type' => 'string',
+                        'default' => 'id',
+                        'sanitize_callback' => 'sanitize_text_field',
+                    ],
+                    'sortDirection' => [
+                        'type' => 'string',
+                        'default' => 'asc',
+                        'enum' => ['asc', 'desc'],
+                    ],
+                ],
+            ]
+        );
+    }
+
+    /**
+     * @unreleased
+     *
+     * @param WP_REST_Request $request
+     *
+     * @return WP_REST_Response
+     */
+    public function handleRequest(WP_REST_Request $request): WP_REST_Response
+    {
+        $query = Event::query()
+            ->orderBy(
+                $request->get_param('sortColumn') ?: 'id',
+                $request->get_param('sortDirection') ?: 'asc'
+            );
+
+        if($search = $request->get_param('search')) {
+            $query
+                ->whereLike('title', "%$search%")
+                ->orWhereLike('description', "%$search%");
+        }
+
+        $events = $query
+            ->paginate(
+                $request->get_param('perPage'),
+                $request->get_param('page')
+            );
+
+        return new WP_REST_Response(
+            [
+                'items' => array_map(
+                    [$this, 'transformEventToRow'],
+                    $events->getAll() ?: []
+                ),
+                'totalItems' => $count = $events->count(),
+                'totalPages' => ceil($count / $request->get_param('perPage')),
+            ]
+        );
+    }
+
+    /**
+     * @unreleased
+     */
+    protected function transformEventToRow(Event $event): array
+    {
+        return [
+            'id' => $event->id,
+            'title' => $this->formatColumnTitle($event),
+            'description' => $event->description,
+            'startDate' => $event->start_datetime->format('m/d/Y g:i a'),
+            'ticketsSold' => $event->eventTickets()->count(),
+        ];
+    }
+
+    /**
+     * @unreleased
+     */
+    protected function formatColumnTitle(Event $event): string
+    {
+        return sprintf(
+            '<a href="%s" aria-label="%s">%s</a>',
+            admin_url("edit.php?post_type=give_forms&page=give-events&view=overview&id=$event->id"),
+            __('View event details', 'give'),
+            $event->title
+        );
+    }
```

Comment:
Hey @kjohnson, are you just scaffolding the endpoint for the Events List Table? If so, we are good to go. Otherwise, we need to stick with the `ListTable` structure that separate some concerns and allows 3rd-party users to manage columns.

Check the files below:
```
Give\Donations\Endpoint\ListDonations.php
Give\Donations\ListTable\DonationsListTable.php
Give\Donations\ListTable\Columns\*.php
```

Reply:
We don't need to introduce extensibility yet. A simple array mapper will work for now.